src/huggingface_hub/hf_api.py

```diff
@@ -1726,20 +1736,138 @@ def move_repo(
             " completed."
         )
 
+    def create_commit(
+        self,
+        repo_id: str,
+        operations: Iterable[CommitOperation],
+        *,
+        commit_summary: str,
+        commit_description: Optional[str] = None,
+        token: Optional[str] = None,
+        repo_type: Optional[str] = None,
+        revision: Optional[str] = None,
+        create_pr: Optional[bool] = None,
+        num_threads: int = 5,
+    ) -> Optional[str]:
+        """
+        Creates a commit in the given repo, deleting & uploading files as needed.
+
+        Args:
+            repo_id (`str`):
+                The repository in which the commit will be created, for example:
+                `"username/custom_transformers"`
+
+            operations (`Iterable` of `CommitOperation`):
+                An iterable of operations to include in the commit, either:
+
+                    - ``CommitOperationAdd`` to upload a file
+                    - ``CommitOperationDelete`` to delete a file
+
+            commit_summary (``str``):
+                The summary of the commit that will be created
+
+            commit_description (``str``, *optional*):
+                The description of the commit that will be created
+
```

Comment:
The difference between these two is not very clear to me. Is `summary` first line and description the rest? Should we rename one of these to `message` for consistency with `git commit -m...`?